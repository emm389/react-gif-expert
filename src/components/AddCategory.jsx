import { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Search Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

// AddCategory.propTypes = {
//     categories: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
//     setCategories: PropTypes.func.isRequired
// }

export default AddCategory
