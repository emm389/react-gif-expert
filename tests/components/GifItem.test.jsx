import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Tests on GifItem', () => {

    const title = 'Dragon Ball';
    const url = 'https://www.genbeta.com/';

    test('should to match with the snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should to display the image with the url and the ALT indicated', () => {
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(alt);

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title)
    });
    
    test('should to display the title in the component', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title))
    });
})
