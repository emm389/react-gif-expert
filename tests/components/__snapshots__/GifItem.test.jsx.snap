// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifItem should to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://www.genbeta.com/"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
